include/miopen/miopen.h

```diff
@@ -4950,6 +4950,61 @@ MIOPEN_EXPORT miopenStatus_t miopenCTCLoss(miopenHandle_t handle,
                                            void* workSpace,
                                            size_t workSpaceSize);
 
+#ifdef MIOPEN_BETA_API
+
+typedef enum
+{
+    MIOPEN_LOSS_REDUCTION_NONE = 0, /*!< output tensor elements are not reduced */
+    MIOPEN_LOSS_REDUCTION_SUM  = 1, /*!< output tensor elements are summed up */
+    MIOPEN_LOSS_REDUCTION_MEAN = 2, /*!< output tensor elements are summed up and divided with total
+                                       number of elements to get mean value */
+} miopenLossReductionMode_t;
+
+/*! @brief Helper function to query the minimum workspace size required by the multilabel margin
+ * loss call
+ *
+ * @param handle                   MIOpen Handle (input)
+ * @param iDesc                    Tensor descriptor for input tensor (input)
+ * @param tDesc                    Tensor descriptor for target tensor (input)
+ * @param oDesc                    Tensor descriptor for output tensor (input)
+ * @param sizeInBytes              Pointer to data to return the minimum workspace size
+ * @return                         miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t
+miopenGetMultilabelMarginLossForwardWorkspaceSize(miopenHandle_t handle,
+                                                  miopenTensorDescriptor_t iDesc,
+                                                  miopenTensorDescriptor_t tDesc,
+                                                  miopenTensorDescriptor_t oDesc,
+                                                  size_t* sizeInBytes);
+
+/*! @brief Execute MultilabelMargin Loss forward layer
+ *
+ * @param handle                   MIOpen handle (input)
+ * @param workspace                Address of the allocated workspace data (input)
+ * @param workspaceSizeInBytes     Size in bytes of the allocated workspace data (input)
+ * @param iDesc                    Tensor descriptor for input tensor (input)
+ * @param i                        Data tensor input (input)
+ * @param tDesc                    Tensor descriptor for target tensor (input)
+ * @param t                        Data tensor target (input)
+ * @param oDesc                    Tensor descriptor for output tensor (input)
+ * @param o                        Data tensor output (output)
+ * @param divisor                  Divisor (input)
+ * @return                         miopenStatus_t
+ */
+
+MIOPEN_EXPORT miopenStatus_t miopenMultilabelMarginLossForward(miopenHandle_t handle,
```

Comment:
Refer to https://github.com/moreh-dev/MIOpen/pull/20#pullrequestreview-2080454568 to receive the mode from the user and calculate the divisor internally.

Reply:
resolved in commit [301a1c6](https://github.com/moreh-dev/MIOpen/pull/25/commits/301a1c66f3bdec977c1d2fdf8203e3451c62bb9c)